internal/pkg/app/handlers_v1.go

```diff
@@ -121,7 +120,7 @@ func (a *APIManager) addRouteHandler(w http.ResponseWriter, r *http.Request) {
 	err = a.routingTableMgr.AddRoute(ctx, &route)
 	if err != nil {
 		log.Ctx(ctx).Error().Str("op", "addRouteHandler").Msg(err.Error())
-		resp := ErrorResponse(err)
+		resp := ErrorResponse(&BadRequestError{err.Error(), err})
```

Comment:
I don't think we can just do a catch-all here. There might be other errors such as db connections error that are not badrequest error. I think we need to either need to check the error returned by AddRoute or have AddRoute return different type of errors. Also, we really should cleanup the sentinel errors in routingTableMgr per our agreement

Reply:
so picky ;)

Reply:
on a more serious note i was also wondering if it really makes sense to raise 4xx (or 5xx) error codes from lower layers that are far away and detached from the rest api layer

Reply:
Personally, I think it is OK to raise 4xx code from lower level to make our code more concise. Maybe we should have a bigger discussion